src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -6308,71 +6308,131 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
         noway_assert(uCntBytes == 0);
 
 #elif defined(TARGET_XARCH)
-        /*
-            Generate the following code:
-
-                lea     edi, [ebp/esp-OFFS]
-                mov     ecx, <size>
-                xor     eax, eax
-                rep     stosd
-         */
+        noway_assert(compiler->getSIMDSupportLevel() >= SIMD_SSE2_Supported);
+        // As we output multiple instructions for SIMD zeroing, we want to balance code size, with throughput
+        // so cap max size at 16 * MAX SIMD length
+        int initMaxSIMDSize = 16 * XMM_REGSIZE_BYTES;
+        if ((untrLclHi - untrLclLo) <= initMaxSIMDSize)
+        {
+            /*
+                Generate the following code:
+                   xor      rax, rax
+                   vxorps   xmm4, xmm4
+                   vmovdqu  xmmword ptr [ebp/esp-OFFS], xmm4
+                   ...
+                   vmovdqu  xmmword ptr [ebp/esp-OFFS], xmm4
+                   mov      qword ptr [ebp/esp-OFFS], rax
+             */
```

Comment:
Please use end-of-line style `//` comments: https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/clr-jit-coding-conventions.md#7.1.1

Reply:
Done